classes/form/settings_form.php

```diff
@@ -108,6 +108,10 @@ public function define_file_transfer_section($mform, $config) {
         $mform->addHelpButton('sizethreshold', 'settings:sizethreshold', 'tool_objectfs');
         $mform->setType("sizethreshold", PARAM_INT);
 
+        $mform->addElement('text', 'batchsize', get_string('settings:batchsize', 'tool_objectfs'));
+        $mform->addHelpButton('batchsize', 'settings:batchsize', 'tool_objectfs');
+        $mform->setType("batchsize", PARAM_INT);
+
```

Comment:
I think you are missing a setDefault() which should also be 1000?

Reply:
Default value is set in `lib.php`:
https://github.com/catalyst/moodle-tool_objectfs/pull/158/commits/cc1b80cec00bb9360a405f59b702e6db4e1785d2#diff-828043dc9f8d10ccd58ff8b96297794fR88

Reply:
I still think the formslib should be told what the default is, even if the default will be set before being loaded into the form. The default value gets shown as a hint in the form UI this way.

Reply:
As discussed we leave it as is.